### PR TITLE
Force a repaint after reset the format

### LINKED ISSUE #753: Set Formatter  option does not work for "static" attributes
Set Formatter  context menu option does not work for attributes that not send events. Even-thought the formatter has changed the widget still showing the value in the old mode. If you force a repaint in the widget (e.g. taurusform) it will be automatic applied.

IMO a repaint should be done after a formatter changed.

### PR DESCRIPTION
**Pull request by cmft**
The "static" attributes (whitout events) do not change the
representation after a Fotmatter changed.

Fix it, calling  getDisplayValue() after reset the format.

Fix #753

### CODE REVIEW

**Comment by cpascual** [comment]
Hi, I pushed a change to docstrings to reflect the new behaviour. 
Apart from that, I think the fix is ok.
Can you please also update the CHANGELOG?

**Comment by cpascual** [comment]
Oops! and also note that a has been broken with this change:

```
======================================================================
FAIL: test_text_7 (taurus.qt.qtgui.display.test.test_tauruslabel.TaurusLabelTest2)
Testing text(expected='ATTR_VALID', model='tango:TangoSchemeTest/unittest/temp-1/double_scalar', maxdepr=1, fgRole='quality')
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/usr/local/lib/python2.7/dist-packages/taurus-4.3.2a0-py2.7.egg/taurus/test/base.py", line 113, in newTest
    return helper(**helper_kwargs)
  File "/usr/local/lib/python2.7/dist-packages/taurus-4.3.2a0-py2.7.egg/taurus/qt/qtgui/display/test/test_tauruslabel.py", line 157, in text
    self.assertEqual(got, expected, msg)
AssertionError: wrong text for "tango:TangoSchemeTest/unittest/temp-1/double_scalar":
 expected: ATTR_VALID
 got: 0
```

File: lib/taurus/qt/qtgui/display/test/test_tauruslabel.py [hunk 1]
```diff
@@ -94,47 +94,47 @@ def test_relativemodelclass(self):
                                    model='tango:' + DEV_NAME + '/double_scalar')
 
 
-@testOldFgroles(fgRole='value', expected='1.23 mm')
-@testOldFgroles(fgRole='w_value', expected='0.00 mm')
-@testOldFgroles(fgRole='state', expected='Ready')
+# @testOldFgroles(fgRole='value', expected='1.23 mm')
```
Comment (cpascual) [inline lib/taurus/qt/qtgui/display/test/test_tauruslabel.py]: I guess you commented out these tests while debugging but you forgot to uncoment them back....

File: lib/taurus/qt/qtgui/display/tauruslabel.py [hunk 2]
```diff
@@ -90,6 +90,8 @@ def _updateForeground(self, label):
         elif fgRole.lower() in ('', 'none'):
             pass
         else:
+            label.modelFragmentName = fgRole
```
Comment (cpascual) [inline lib/taurus/qt/qtgui/display/tauruslabel.py]: I do not like the idea of altering the label.modelFragmentName every time that you call _updateForeground on its controller (this is a side-efect and it is asking for future trouble, IMHO).
At the very least, I think that the original fragment name should be resored just after calling resetFormat.

...and even that would look quite ugly.

Isn't there any more elegant solution?

File: lib/taurus/qt/qtgui/display/test/test_tauruslabel.py (hunk 1 above)
Comment (cmft) [reply lib/taurus/qt/qtgui/display/test/test_tauruslabel.py]: You are right, I forget to remove the "kk" commit.

File: lib/taurus/qt/qtgui/display/tauruslabel.py (hunk 2 above)
Comment (cmft) [reply lib/taurus/qt/qtgui/display/tauruslabel.py]: > Isn't there any more elegant solution?

I did not find it, the other solution is at the taurusform level calling  crl.update  method after set a new Format

**Comment by cmft** [comment]
I close this PR since the necessary workaround can introduce more troubles. For the moment the manual  resizing of the GUI solves the problem

**Pull request closed without merging**